api/serializers/enrollment_test.py

```diff
@@ -45,8 +52,43 @@ class Meta:
         fields = (
             "student",
             "answers",
-            "resulting_level",
         )
 
     # TODO Think about some validation, e.g. answers should be unique for each question
     #  and should be in options for this question
+
+
+class EnrollmentTestResultLevelSerializer(serializers.ModelSerializer[EnrollmentTestResult]):
+    """A serializer used to get level of language based on how many answers are correct."""
+
+    resulting_level = serializers.SerializerMethodField()
+
+    class Meta:
+        model = EnrollmentTestResult
+        fields = ["answers", "resulting_level"]
+
+    def validate(self, attrs: dict[str, Any]) -> dict[str, Any]:
+        if len(attrs["answers"]) not in ENROLLMENT_TEST_LEVEL_THRESHOLDS_FOR_NUMBER_OF_QUESTIONS:
+            raise serializers.ValidationError(
+                f"Enrollment test with {len(attrs['answers'])} questions is not supported"
```

Comment:
`len(attrs["answers"])` used twice here, so it's better to put it into variable

Reply:
Fixed